inc/classes/class-service-worker.php

```diff
@@ -107,23 +107,35 @@ public function cache_images( \WP_Service_Worker_Scripts $scripts ) {
 	 */
 	public function precache_latest_blog_posts( \WP_Service_Worker_Scripts $scripts ) {
 
-		$recent_posts = new \WP_Query(
-			array(
-				'post_type'              => 'post',
-				'post_status'            => 'publish',
-				'fields'                 => 'ids',
-				'posts_per_page'         => 10,
-				'no_found_rows'          => true,
-				'update_post_meta_cache' => false,
-				'update_post_term_cache' => false,
-			)
-		);
+		$cache_key = 'rt_pwa_extensions_precacge_latest_posts';
+
+		$recent_posts = wp_cache_get( $cache_key );
+
+		if ( empty( $recent_posts ) ) {
+
+			$recent_posts = new \WP_Query(
+				array(
+					'post_type'              => 'post',
+					'post_status'            => 'publish',
+					'fields'                 => 'ids',
+					'posts_per_page'         => 10,
+					'no_found_rows'          => true,
+					'update_post_meta_cache' => false,
+					'update_post_term_cache' => false,
+				)
+			);
+
+			if ( empty( $recent_posts->posts ) || ! is_array( $recent_posts->posts ) ) {
+				return;
+			}
+
+			$recent_posts = $recent_posts->posts;
+
+			wp_cache_set( $cache_key, $recent_posts, '', 10 * MINUTE_IN_SECONDS );
```

Comment:
@pradeep910 FYI -- We need to cache results for max 10 mins so it new latest posts can precache and we don't need to delete the cache when new post created. Same I did for menu links.